geopandas/io/file.py

```diff
@@ -197,15 +198,33 @@ def to_file(*args, **kwargs):
     return _to_file(*args, **kwargs)
 
 
+_DRIVER_EXTENSION_MAP = {
+    ".shp": "ESRI Shapefile",
+    "": "ESRI Shapefile",  # shapefile folder
+    ".gpkg": "GPKG",
+    ".geojson": "GeoJSON",
+    ".json": "GeoJSON",
+}
+
+
+def _detect_driver(path):
+    """
+    Attempt to auto-detect driver based on the extension
+    """
+    try:
+        # in case the path is a file handle
+        path = path.name
+    except AttributeError:
+        pass
+    try:
+        return _DRIVER_EXTENSION_MAP[Path(path).suffix.lower()]
+    except KeyError:
+        # pass to fiona to handle the missing driver
+        return None
```

Comment:
I think this is fine if we use the complete dict I proposed above. I would maybe add a warning here explaining how to fix the issue because fiona's error shows `DriverError: no driver` only.

Reply:
Sounds like a good idea 👍